pgmpy/base/DAG.py

```diff
@@ -11,6 +11,7 @@
 from pgmpy.base import UndirectedGraph
 from pgmpy.global_vars import logger
 from pgmpy.independencies import Independencies
+from pgmpy.metrics.metrics import correlation_score, fisher_c, structure_score
```

Comment:
You might need to put this import inside the validate method. Could be leading to the circular import because of which the tests are failing.